test member remove fails on CI

```
=== RUN   TestMemberRemove
2016-03-08 05:51:17.880602 I | etcdserver: name = node8162987141596479378
2016-03-08 05:51:17.880639 I | etcdserver: data dir = /tmp/etcd988753435
2016-03-08 05:51:17.880655 I | etcdserver: member dir = /tmp/etcd988753435/member
2016-03-08 05:51:17.880663 I | etcdserver: heartbeat = 10ms
2016-03-08 05:51:17.880670 I | etcdserver: election = 100ms
2016-03-08 05:51:17.880677 I | etcdserver: snapshot count = 0
2016-03-08 05:51:17.880692 I | etcdserver: advertise client URLs = http://127.0.0.1:20194
2016-03-08 05:51:17.880700 I | etcdserver: initial advertise peer URLs = http://127.0.0.1:20193
2016-03-08 05:51:17.880732 I | etcdserver: initial cluster = node5467211863980644661=http://127.0.0.1:20189,node598356869608421287=http://127.0.0.1:20191,node8162987141596479378=http://127.0.0.1:20193
2016-03-08 05:51:17.880756 I | etcdserver: name = node598356869608421287
2016-03-08 05:51:17.880773 I | etcdserver: data dir = /tmp/etcd901099692
2016-03-08 05:51:17.880782 I | etcdserver: member dir = /tmp/etcd901099692/member
2016-03-08 05:51:17.880791 I | etcdserver: heartbeat = 10ms
2016-03-08 05:51:17.880798 I | etcdserver: election = 100ms
2016-03-08 05:51:17.880804 I | etcdserver: snapshot count = 0
2016-03-08 05:51:17.880816 I | etcdserver: advertise client URLs = http://127.0.0.1:20192
2016-03-08 05:51:17.880824 I | etcdserver: initial advertise peer URLs = http://127.0.0.1:20191
2016-03-08 05:51:17.880844 I | etcdserver: initial cluster = node5467211863980644661=http://127.0.0.1:20189,node598356869608421287=http://127.0.0.1:20191,node8162987141596479378=http://127.0.0.1:20193
2016-03-08 05:51:17.881067 I | etcdserver: name = node5467211863980644661
2016-03-08 05:51:17.881099 I | etcdserver: data dir = /tmp/etcd573352513
2016-03-08 05:51:17.881124 I | etcdserver: member dir = /tmp/etcd573352513/member
2016-03-08 05:51:17.881137 I | etcdserver: heartbeat = 10ms
2016-03-08 05:51:17.881149 I | etcdserver: election = 100ms
2016-03-08 05:51:17.881160 I | etcdserver: snapshot count = 0
2016-03-08 05:51:17.881176 I | etcdserver: advertise client URLs = http://127.0.0.1:20190
2016-03-08 05:51:17.881192 I | etcdserver: initial advertise peer URLs = http://127.0.0.1:20189
2016-03-08 05:51:17.881217 I | etcdserver: initial cluster = node5467211863980644661=http://127.0.0.1:20189,node598356869608421287=http://127.0.0.1:20191,node8162987141596479378=http://127.0.0.1:20193
2016-03-08 05:51:17.881359 I | etcdserver: starting member e9926f6404e5a7c3 in cluster 3ed5d76142ee866f
2016-03-08 05:51:17.881439 I | raft: e9926f6404e5a7c3 became follower at term 0
2016-03-08 05:51:17.881474 I | raft: newRaft e9926f6404e5a7c3 [peers: [], term: 0, commit: 0, applied: 0, lastindex: 0, lastterm: 0]
2016-03-08 05:51:17.881497 I | raft: e9926f6404e5a7c3 became follower at term 1
2016-03-08 05:51:17.881595 I | etcdserver: starting member 6a250c1a267bbf2c in cluster 3ed5d76142ee866f
2016-03-08 05:51:17.881649 I | raft: 6a250c1a267bbf2c became follower at term 0
2016-03-08 05:51:17.881671 I | raft: newRaft 6a250c1a267bbf2c [peers: [], term: 0, commit: 0, applied: 0, lastindex: 0, lastterm: 0]
2016-03-08 05:51:17.881685 I | raft: 6a250c1a267bbf2c became follower at term 1
2016-03-08 05:51:17.884987 I | etcdserver: set snapshot count to default 10000
2016-03-08 05:51:17.885048 I | etcdserver: starting server... [version: 2.3.0-alpha.1+git, cluster version: to_be_decided]
2016-03-08 05:51:17.885085 I | etcdserver: set snapshot count to default 10000
2016-03-08 05:51:17.885128 I | etcdserver: starting server... [version: 2.3.0-alpha.1+git, cluster version: to_be_decided]
2016-03-08 05:51:17.885429 I | etcdserver: starting member 331b18f8e514e721 in cluster 3ed5d76142ee866f
2016-03-08 05:51:17.885525 I | raft: 331b18f8e514e721 became follower at term 0
2016-03-08 05:51:17.885567 I | raft: newRaft 331b18f8e514e721 [peers: [], term: 0, commit: 0, applied: 0, lastindex: 0, lastterm: 0]
2016-03-08 05:51:17.885604 I | raft: 331b18f8e514e721 became follower at term 1
2016-03-08 05:51:17.907046 N | etcdserver: added member 331b18f8e514e721 [http://127.0.0.1:20191] to cluster 3ed5d76142ee866f
2016-03-08 05:51:17.907222 N | etcdserver: added local member 6a250c1a267bbf2c [http://127.0.0.1:20189] to cluster 3ed5d76142ee866f
2016-03-08 05:51:17.907418 N | etcdserver: added member e9926f6404e5a7c3 [http://127.0.0.1:20193] to cluster 3ed5d76142ee866f
2016-03-08 05:51:17.907479 N | etcdserver: added member 331b18f8e514e721 [http://127.0.0.1:20191] to cluster 3ed5d76142ee866f
2016-03-08 05:51:17.907656 N | etcdserver: added member 6a250c1a267bbf2c [http://127.0.0.1:20189] to cluster 3ed5d76142ee866f
2016-03-08 05:51:17.907788 I | rafthttp: the connection with e9926f6404e5a7c3 became active
2016-03-08 05:51:17.907882 N | etcdserver: added local member e9926f6404e5a7c3 [http://127.0.0.1:20193] to cluster 3ed5d76142ee866f
2016-03-08 05:51:17.910371 I | rafthttp: the connection with 6a250c1a267bbf2c became active
2016-03-08 05:51:17.911502 I | etcdserver: set snapshot count to default 10000
2016-03-08 05:51:17.911532 I | etcdserver: starting server... [version: 2.3.0-alpha.1+git, cluster version: to_be_decided]
2016-03-08 05:51:17.912605 I | rafthttp: the connection with 331b18f8e514e721 became active
2016-03-08 05:51:17.912674 I | rafthttp: the connection with 6a250c1a267bbf2c became active
2016-03-08 05:51:17.913363 I | rafthttp: the connection with e9926f6404e5a7c3 became active
2016-03-08 05:51:17.913518 I | rafthttp: the connection with 331b18f8e514e721 became active
2016-03-08 05:51:17.913817 N | etcdserver: added local member 331b18f8e514e721 [http://127.0.0.1:20191] to cluster 3ed5d76142ee866f
2016-03-08 05:51:17.913875 N | etcdserver: added member 6a250c1a267bbf2c [http://127.0.0.1:20189] to cluster 3ed5d76142ee866f
2016-03-08 05:51:17.913920 N | etcdserver: added member e9926f6404e5a7c3 [http://127.0.0.1:20193] to cluster 3ed5d76142ee866f
2016-03-08 05:51:17.921789 I | raft: e9926f6404e5a7c3 is starting a new election at term 1
2016-03-08 05:51:17.921846 I | raft: e9926f6404e5a7c3 became candidate at term 2
2016-03-08 05:51:17.921851 I | raft: e9926f6404e5a7c3 received vote from e9926f6404e5a7c3 at term 2
2016-03-08 05:51:17.921858 I | raft: e9926f6404e5a7c3 [logterm: 1, index: 3] sent vote request to 331b18f8e514e721 at term 2
2016-03-08 05:51:17.921864 I | raft: e9926f6404e5a7c3 [logterm: 1, index: 3] sent vote request to 6a250c1a267bbf2c at term 2
2016-03-08 05:51:17.922030 I | raft: 331b18f8e514e721 [term: 1] received a MsgVote message with higher term from e9926f6404e5a7c3 [term: 2]
2016-03-08 05:51:17.922042 I | raft: 331b18f8e514e721 became follower at term 2
2016-03-08 05:51:17.922060 I | raft: 331b18f8e514e721 [logterm: 1, index: 3, vote: 0] voted for e9926f6404e5a7c3 [logterm: 1, index: 3] at term 2
2016-03-08 05:51:17.922098 I | raft: 6a250c1a267bbf2c [term: 1] received a MsgVote message with higher term from e9926f6404e5a7c3 [term: 2]
2016-03-08 05:51:17.922112 I | raft: 6a250c1a267bbf2c became follower at term 2
2016-03-08 05:51:17.922127 I | raft: 6a250c1a267bbf2c [logterm: 1, index: 3, vote: 0] voted for e9926f6404e5a7c3 [logterm: 1, index: 3] at term 2
2016-03-08 05:51:17.922146 I | raft: e9926f6404e5a7c3 received vote from 331b18f8e514e721 at term 2
2016-03-08 05:51:17.922163 I | raft: e9926f6404e5a7c3 [quorum:2] has received 2 votes and 0 vote rejections
2016-03-08 05:51:17.922172 I | raft: e9926f6404e5a7c3 became leader at term 2
2016-03-08 05:51:17.922179 I | raft: raft.node: e9926f6404e5a7c3 elected leader e9926f6404e5a7c3 at term 2
2016-03-08 05:51:17.922265 I | raft: raft.node: 331b18f8e514e721 elected leader e9926f6404e5a7c3 at term 2
2016-03-08 05:51:17.922283 I | raft: raft.node: 6a250c1a267bbf2c elected leader e9926f6404e5a7c3 at term 2
2016-03-08 05:51:17.922876 I | etcdserver: published {Name:node8162987141596479378 ClientURLs:[http://127.0.0.1:20194]} to cluster 3ed5d76142ee866f
2016-03-08 05:51:17.923100 I | etcdserver: published {Name:node598356869608421287 ClientURLs:[http://127.0.0.1:20192]} to cluster 3ed5d76142ee866f
2016-03-08 05:51:17.923162 I | etcdserver: published {Name:node5467211863980644661 ClientURLs:[http://127.0.0.1:20190]} to cluster 3ed5d76142ee866f
2016-03-08 05:51:17.923275 I | etcdserver: setting up the initial cluster version to 2.3
2016-03-08 05:51:17.923467 N | etcdserver: set the initial cluster version to 2.3
2016-03-08 05:51:17.923523 N | etcdserver: set the initial cluster version to 2.3
2016-03-08 05:51:17.923540 N | etcdserver: set the initial cluster version to 2.3
2016-03-08 05:51:17.935711 N | etcdserver: removed member 331b18f8e514e721 from cluster 3ed5d76142ee866f
2016-03-08 05:51:17.936174 E | rafthttp: failed to read 331b18f8e514e721 on stream Message (net/http: request canceled)
2016-03-08 05:51:17.936211 I | rafthttp: the connection with 331b18f8e514e721 became inactive
2016-03-08 05:51:17.936943 N | etcdserver: removed member 331b18f8e514e721 from cluster 3ed5d76142ee866f
2016-03-08 05:51:18.036260 W | rafthttp: rejected the stream from peer 331b18f8e514e721 since it was removed
2016-03-08 05:51:18.036298 W | rafthttp: rejected the stream from peer 331b18f8e514e721 since it was removed
2016-03-08 05:51:18.036396 E | rafthttp: failed to dial e9926f6404e5a7c3 on stream Message (the member has been permanently removed from the cluster)
2016-03-08 05:51:18.036411 I | rafthttp: the connection with e9926f6404e5a7c3 became inactive
2016-03-08 05:51:18.036444 E | etcdserver: the member has been permanently removed from the cluster
2016-03-08 05:51:18.036452 I | etcdserver: the data-dir used by this member must be removed.
2016-03-08 05:51:18.036825 W | rafthttp: rejected the stream from peer 331b18f8e514e721 since it was removed
2016-03-08 05:51:18.036975 E | rafthttp: failed to dial 6a250c1a267bbf2c on stream MsgApp v2 (the member has been permanently removed from the cluster)
2016-03-08 05:51:18.036991 I | rafthttp: the connection with 6a250c1a267bbf2c became inactive
2016-03-08 05:51:18.038703 E | rafthttp: failed to read e9926f6404e5a7c3 on stream MsgApp v2 (net/http: request canceled)
2016-03-08 05:51:18.038723 I | rafthttp: the connection with e9926f6404e5a7c3 became inactive
--- FAIL: TestMemberRemove (0.18s)
	cluster_test.go:78: failed to remove member rpc error: code = 13 desc = "etcdserver: server stopped"
```

Also seeing a double-close panic:

```
=== RUN   TestMemberRemove
2016-03-08 10:12:34.097886 I | etcdserver: name = node5690509182857642078
2016-03-08 10:12:34.097914 I | etcdserver: data dir = /Volumes/Ram Disk/etcd100980391
2016-03-08 10:12:34.097921 I | etcdserver: member dir = /Volumes/Ram Disk/etcd100980391/member
2016-03-08 10:12:34.097925 I | etcdserver: heartbeat = 10ms
2016-03-08 10:12:34.097987 I | etcdserver: election = 100ms
2016-03-08 10:12:34.097998 I | etcdserver: snapshot count = 0
2016-03-08 10:12:34.098011 I | etcdserver: advertise client URLs = http://127.0.0.1:20078
2016-03-08 10:12:34.098021 I | etcdserver: initial advertise peer URLs = http://127.0.0.1:20077
2016-03-08 10:12:34.098035 I | etcdserver: initial cluster = node4471102156321863537=http://127.0.0.1:20075,node5649691784882221040=http://127.0.0.1:20073,node5690509182857642078=http://127.0.0.1:20077
2016-03-08 10:12:34.098133 I | etcdserver: name = node4471102156321863537
2016-03-08 10:12:34.098147 I | etcdserver: data dir = /Volumes/Ram Disk/etcd135505000
2016-03-08 10:12:34.098155 I | etcdserver: member dir = /Volumes/Ram Disk/etcd135505000/member
2016-03-08 10:12:34.098162 I | etcdserver: heartbeat = 10ms
2016-03-08 10:12:34.098169 I | etcdserver: election = 100ms
2016-03-08 10:12:34.098175 I | etcdserver: snapshot count = 0
2016-03-08 10:12:34.098185 I | etcdserver: advertise client URLs = http://127.0.0.1:20076
2016-03-08 10:12:34.098194 I | etcdserver: initial advertise peer URLs = http://127.0.0.1:20075
2016-03-08 10:12:34.098220 I | etcdserver: initial cluster = node4471102156321863537=http://127.0.0.1:20075,node5649691784882221040=http://127.0.0.1:20073,node5690509182857642078=http://127.0.0.1:20077
2016-03-08 10:12:34.098290 I | etcdserver: name = node5649691784882221040
2016-03-08 10:12:34.098309 I | etcdserver: data dir = /Volumes/Ram Disk/etcd228171245
2016-03-08 10:12:34.098330 I | etcdserver: member dir = /Volumes/Ram Disk/etcd228171245/member
2016-03-08 10:12:34.098338 I | etcdserver: heartbeat = 10ms
2016-03-08 10:12:34.098348 I | etcdserver: election = 100ms
2016-03-08 10:12:34.098356 I | etcdserver: snapshot count = 0
2016-03-08 10:12:34.098375 I | etcdserver: advertise client URLs = http://127.0.0.1:20074
2016-03-08 10:12:34.098396 I | etcdserver: initial advertise peer URLs = http://127.0.0.1:20073
2016-03-08 10:12:34.098432 I | etcdserver: initial cluster = node4471102156321863537=http://127.0.0.1:20075,node5649691784882221040=http://127.0.0.1:20073,node5690509182857642078=http://127.0.0.1:20077
2016-03-08 10:12:34.099049 I | etcdserver: starting member 91a88c050087cd6b in cluster 338c4f36ac463586
2016-03-08 10:12:34.099088 I | raft: 91a88c050087cd6b became follower at term 0
2016-03-08 10:12:34.099099 I | raft: newRaft 91a88c050087cd6b [peers: [], term: 0, commit: 0, applied: 0, lastindex: 0, lastterm: 0]
2016-03-08 10:12:34.099103 I | raft: 91a88c050087cd6b became follower at term 1
2016-03-08 10:12:34.099389 I | etcdserver: starting member 10884cb531021677 in cluster 338c4f36ac463586
2016-03-08 10:12:34.099424 I | raft: 10884cb531021677 became follower at term 0
2016-03-08 10:12:34.099438 I | raft: newRaft 10884cb531021677 [peers: [], term: 0, commit: 0, applied: 0, lastindex: 0, lastterm: 0]
2016-03-08 10:12:34.099453 I | raft: 10884cb531021677 became follower at term 1
2016-03-08 10:12:34.099584 I | etcdserver: starting member aa718eac5b4a3bc2 in cluster 338c4f36ac463586
2016-03-08 10:12:34.099633 I | raft: aa718eac5b4a3bc2 became follower at term 0
2016-03-08 10:12:34.099651 I | raft: newRaft aa718eac5b4a3bc2 [peers: [], term: 0, commit: 0, applied: 0, lastindex: 0, lastterm: 0]
2016-03-08 10:12:34.099661 I | raft: aa718eac5b4a3bc2 became follower at term 1
2016-03-08 10:12:34.103131 I | etcdserver: set snapshot count to default 10000
2016-03-08 10:12:34.103528 I | etcdserver: starting server... [version: 2.3.0-alpha.1+git, cluster version: to_be_decided]
2016-03-08 10:12:34.107275 E | etcdserver: cannot monitor file descriptor usage (cannot get FDUsage on darwin)
2016-03-08 10:12:34.108489 N | etcdserver: added member 10884cb531021677 [http://127.0.0.1:20075] to cluster 338c4f36ac463586
2016-03-08 10:12:34.108643 N | etcdserver: added local member 91a88c050087cd6b [http://127.0.0.1:20077] to cluster 338c4f36ac463586
2016-03-08 10:12:34.109421 N | etcdserver: added member aa718eac5b4a3bc2 [http://127.0.0.1:20073] to cluster 338c4f36ac463586
2016-03-08 10:12:34.113538 I | etcdserver: set snapshot count to default 10000
2016-03-08 10:12:34.113576 I | etcdserver: starting server... [version: 2.3.0-alpha.1+git, cluster version: to_be_decided]
2016-03-08 10:12:34.114158 I | rafthttp: the connection with 10884cb531021677 became active
2016-03-08 10:12:34.114181 I | rafthttp: the connection with 91a88c050087cd6b became active
2016-03-08 10:12:34.114226 E | etcdserver: cannot monitor file descriptor usage (cannot get FDUsage on darwin)
2016-03-08 10:12:34.114256 I | rafthttp: the connection with aa718eac5b4a3bc2 became active
2016-03-08 10:12:34.114413 I | rafthttp: the connection with 91a88c050087cd6b became active
2016-03-08 10:12:34.115183 N | etcdserver: added member 10884cb531021677 [http://127.0.0.1:20075] to cluster 338c4f36ac463586
2016-03-08 10:12:34.115276 N | etcdserver: added member 91a88c050087cd6b [http://127.0.0.1:20077] to cluster 338c4f36ac463586
2016-03-08 10:12:34.115305 I | etcdserver: set snapshot count to default 10000
2016-03-08 10:12:34.115339 I | etcdserver: starting server... [version: 2.3.0-alpha.1+git, cluster version: to_be_decided]
2016-03-08 10:12:34.115495 E | etcdserver: cannot monitor file descriptor usage (cannot get FDUsage on darwin)
2016-03-08 10:12:34.115509 N | etcdserver: added local member aa718eac5b4a3bc2 [http://127.0.0.1:20073] to cluster 338c4f36ac463586
2016-03-08 10:12:34.115741 N | etcdserver: added local member 10884cb531021677 [http://127.0.0.1:20075] to cluster 338c4f36ac463586
2016-03-08 10:12:34.115848 N | etcdserver: added member 91a88c050087cd6b [http://127.0.0.1:20077] to cluster 338c4f36ac463586
2016-03-08 10:12:34.115932 N | etcdserver: added member aa718eac5b4a3bc2 [http://127.0.0.1:20073] to cluster 338c4f36ac463586
2016-03-08 10:12:34.116111 I | rafthttp: the connection with aa718eac5b4a3bc2 became active
2016-03-08 10:12:34.116195 I | rafthttp: the connection with 10884cb531021677 became active
2016-03-08 10:12:34.120393 I | raft: 10884cb531021677 is starting a new election at term 1
2016-03-08 10:12:34.120468 I | raft: 10884cb531021677 became candidate at term 2
2016-03-08 10:12:34.120474 I | raft: 10884cb531021677 received vote from 10884cb531021677 at term 2
2016-03-08 10:12:34.120482 I | raft: 10884cb531021677 [logterm: 1, index: 3] sent vote request to 91a88c050087cd6b at term 2
2016-03-08 10:12:34.120493 I | raft: 10884cb531021677 [logterm: 1, index: 3] sent vote request to aa718eac5b4a3bc2 at term 2
2016-03-08 10:12:34.120751 I | raft: aa718eac5b4a3bc2 [term: 1] received a MsgVote message with higher term from 10884cb531021677 [term: 2]
2016-03-08 10:12:34.120767 I | raft: aa718eac5b4a3bc2 became follower at term 2
2016-03-08 10:12:34.120783 I | raft: aa718eac5b4a3bc2 [logterm: 1, index: 3, vote: 0] voted for 10884cb531021677 [logterm: 1, index: 3] at term 2
2016-03-08 10:12:34.120836 I | raft: 91a88c050087cd6b [term: 1] received a MsgVote message with higher term from 10884cb531021677 [term: 2]
2016-03-08 10:12:34.120869 I | raft: 91a88c050087cd6b became follower at term 2
2016-03-08 10:12:34.120890 I | raft: 91a88c050087cd6b [logterm: 1, index: 3, vote: 0] voted for 10884cb531021677 [logterm: 1, index: 3] at term 2
2016-03-08 10:12:34.120970 I | raft: 10884cb531021677 received vote from aa718eac5b4a3bc2 at term 2
2016-03-08 10:12:34.120986 I | raft: 10884cb531021677 [quorum:2] has received 2 votes and 0 vote rejections
2016-03-08 10:12:34.121003 I | raft: 10884cb531021677 became leader at term 2
2016-03-08 10:12:34.121031 I | raft: raft.node: 10884cb531021677 elected leader 10884cb531021677 at term 2
2016-03-08 10:12:34.121217 I | raft: raft.node: 91a88c050087cd6b elected leader 10884cb531021677 at term 2
2016-03-08 10:12:34.121237 I | raft: raft.node: aa718eac5b4a3bc2 elected leader 10884cb531021677 at term 2
2016-03-08 10:12:34.122374 I | etcdserver: published {Name:node5690509182857642078 ClientURLs:[http://127.0.0.1:20078]} to cluster 338c4f36ac463586
2016-03-08 10:12:34.122394 I | etcdserver: published {Name:node4471102156321863537 ClientURLs:[http://127.0.0.1:20076]} to cluster 338c4f36ac463586
2016-03-08 10:12:34.123086 I | etcdserver: setting up the initial cluster version to 2.3
2016-03-08 10:12:34.123261 I | etcdserver: published {Name:node5649691784882221040 ClientURLs:[http://127.0.0.1:20074]} to cluster 338c4f36ac463586
2016-03-08 10:12:34.123561 N | etcdserver: set the initial cluster version to 2.3
2016-03-08 10:12:34.123728 N | etcdserver: set the initial cluster version to 2.3
2016-03-08 10:12:34.123774 N | etcdserver: set the initial cluster version to 2.3
2016-03-08 10:12:34.142412 E | rafthttp: failed to read 10884cb531021677 on stream MsgApp v2 (net/http: request canceled)
2016-03-08 10:12:34.142442 I | rafthttp: the connection with 10884cb531021677 became inactive
2016-03-08 10:12:34.142458 E | rafthttp: failed to read 10884cb531021677 on stream MsgApp v2 (net/http: request canceled)
2016-03-08 10:12:34.142474 I | rafthttp: the connection with 10884cb531021677 became inactive
2016-03-08 10:12:34.142504 N | etcdserver: removed member 10884cb531021677 from cluster 338c4f36ac463586
2016-03-08 10:12:34.143153 E | rafthttp: failed to read 91a88c050087cd6b on stream MsgApp v2 (net/http: request canceled)
2016-03-08 10:12:34.143169 I | rafthttp: the connection with 91a88c050087cd6b became inactive
panic: close of closed channel

goroutine 3826 [running]:
github.com/coreos/etcd/rafthttp.(*peer).stop(0xc8201f01e0)
        /Users/anthony/go/src/github.com/coreos/etcd/rafthttp/peer.go:230 +0x2c
github.com/coreos/etcd/rafthttp.(*Transport).removePeer(0xc822b9c360, 0x10884cb531021677)
        /Users/anthony/go/src/github.com/coreos/etcd/rafthttp/transport.go:256 +0x94
github.com/coreos/etcd/rafthttp.(*Transport).RemovePeer(0xc822b9c360, 0x10884cb531021677)
        /Users/anthony/go/src/github.com/coreos/etcd/rafthttp/transport.go:242 +0x84
github.com/coreos/etcd/etcdserver.(*EtcdServer).applyConfChange(0xc821e72380, 0x167753576e36b403, 0x1, 0x10884cb531021677, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /Users/anthony/go/src/github.com/coreos/etcd/etcdserver/server.go:1132 +0x95c
github.com/coreos/etcd/etcdserver.(*EtcdServer).apply(0xc821e72380, 0xc8200e7008, 0x1, 0x7, 0xc820108400, 0x0, 0x0)
        /Users/anthony/go/src/github.com/coreos/etcd/etcdserver/server.go:1021 +0x6c6
github.com/coreos/etcd/etcdserver.(*EtcdServer).applyEntries(0xc821e72380, 0xc820108400, 0xc8200a8dc0)
        /Users/anthony/go/src/github.com/coreos/etcd/etcdserver/server.go:652 +0x2cc
github.com/coreos/etcd/etcdserver.(*EtcdServer).applyAll(0xc821e72380, 0xc820108400, 0xc8200a8dc0)
        /Users/anthony/go/src/github.com/coreos/etcd/etcdserver/server.go:561 +0x78
github.com/coreos/etcd/etcdserver.(*EtcdServer).run.func2(0x1283d30, 0xc8201083c0)
        /Users/anthony/go/src/github.com/coreos/etcd/etcdserver/server.go:541 +0x32
github.com/coreos/etcd/pkg/schedule.(*fifo).run(0xc820125200)
        /Users/anthony/go/src/github.com/coreos/etcd/pkg/schedule/schedule.go:160 +0x322
created by github.com/coreos/etcd/pkg/schedule.NewFIFOScheduler
        /Users/anthony/go/src/github.com/coreos/etcd/pkg/schedule/schedule.go:71 +0x275
```

--- a/clientv3/integration/cluster_test.go
+++ b/clientv3/integration/cluster_test.go
@@ -22,6 +22,7 @@ import (
 	"github.com/coreos/etcd/clientv3"
 	"github.com/coreos/etcd/integration"
 	"github.com/coreos/etcd/pkg/testutil"
+	"github.com/coreos/etcd/pkg/types"
 )
 
 func TestMemberList(t *testing.T) {
@@ -73,7 +74,18 @@ func TestMemberRemove(t *testing.T) {
 		t.Fatalf("failed to list member %v", err)
 	}
 
-	_, err = capi.MemberRemove(context.Background(), resp.Members[0].ID)
+	rmvID := resp.Members[0].ID
+	// indexes in capi member list don't necessarily match cluster member list;
+	// find member that is not the client to remove
+	for _, m := range resp.Members {
+		mURLs, _ := types.NewURLs(m.PeerURLs)
+		if !reflect.DeepEqual(mURLs, clus.Members[1].ServerConfig.PeerURLs) {
+			rmvID = m.ID
+			break
+		}
+	}
+
+	_, err = capi.MemberRemove(context.Background(), rmvID)
 	if err != nil {
 		t.Fatalf("failed to remove member %v", err)
 	}

